### PR TITLE
506966 Fetch With using env variables

### PR DESCRIPTION
**Pull request by CobyPear**
Set fetchWith using env variables in preparation for nextjs Yeoman generator

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: samples/nextjs/src/lib/dictionary-service-factory.ts [hunk 1]
```diff
@@ -1,20 +1,44 @@
-import { DictionaryService, GraphQLDictionaryService } from '@sitecore-jss/sitecore-jss-nextjs';
+import {
+  DictionaryService,
+  RestDictionaryService,
+  GraphQLDictionaryService,
+} from '@sitecore-jss/sitecore-jss-nextjs';
 import config from 'temp/config';
 
-export class DictionaryServiceFactory {
```
Comment (ambrauer) [inline samples/nextjs/src/lib/dictionary-service-factory.ts:4]: I think we should stick to the single factory `DictionaryServiceFactory` (following the factory pattern, it's a factory that creates a `DictionaryService` instance). The check for environment variable / returning appropriate implementation can happen inside the `create()`.

**Review by ambrauer — CHANGES_REQUESTED** [review]

File: samples/nextjs/src/lib/dictionary-service-factory.ts (hunk 1 above)
Comment (CobyPear) [reply samples/nextjs/src/lib/dictionary-service-factory.ts:4]: That makes a lot more sense! Will change.

File: samples/nextjs/.env [hunk 2]
```diff
@@ -36,6 +36,9 @@ SITECORE_API_HOST=
 # the resolved Sitecore API hostname + the `graphQLEndpointPath` defined in your `package.json`.
 GRAPH_QL_ENDPOINT=
 
+# Hardcoded FETCH_WITH env variable to prepare for yeoman generators
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply samples/nextjs/.env:39]: Updated

File: samples/nextjs/src/lib/dictionary-service-factory.ts [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
-      apiKey: config.sitecoreApiKey,
-      siteName: config.jssAppName,
-      /*
+    return process.env.FETCH_WITH === 'graphql'
+      ? new GraphQLDictionaryService({
+          endpoint: config.graphQLEndpoint,
+          apiKey: config.sitecoreApiKey,
+          siteName: config.jssAppName,
+          /*
       The Dictionary Service needs a root item ID in order to fetch dictionary phrases for the current
```
Comment (ambrauer) [inline samples/nextjs/src/lib/dictionary-service-factory.ts:16]: Indentation looks off with the comment here. Otherwise, looks good!

**Review by ambrauer — APPROVED** [review]

**Pull request merged**